packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -49,6 +53,54 @@ contract SphinxUtils is SphinxConstants, StdUtils {
         return _data[_start:_end];
     }
 
+    function checkAccesses(Vm.AccountAccess[] memory accountAccesses) internal view returns (bool) {
+        // If there aren't two calls, (one to the deployment proxy, and another to deploy the contract)
+        // then return false.
+        if (accountAccesses.length < 2) {
+            return false;
+        }
+
+        // If the first access does not record calling deterministic deployment proxy, then return false.
+        if (accountAccesses[0].account != DETERMINISTIC_DEPLOYMENT_PROXY) {
+            return false;
+        }
+
+        // If the second access does not record deploying the contract to the correct address, then return false.
+        address expectedAddress = vm.computeCreate2Address(
+            0,
+            keccak256(type(SphinxForkCheck).creationCode),
+            DETERMINISTIC_DEPLOYMENT_PROXY
+        );
+        if (accountAccesses[1].account != expectedAddress) {
+            return false;
+        }
+
+        // If the second access did not come from the deterministic deployment proxy, then return false
+        if (accountAccesses[1].accessor != DETERMINISTIC_DEPLOYMENT_PROXY) {
+            return false;
+        }
+
+        // If the deployed code at the calculated address is incorrect, then return false.
+        // This confirms the deterministic deployment proxy is in fact being used for the
+        // internal simulation.
+        if (
+            keccak256(address(expectedAddress).code) != keccak256(type(SphinxForkCheck).runtimeCode)
+        ) {
+            return false;
+        }
+
+        return true;
+    }
+
+    function checkFork(address safeAddress) public returns (bool) {
+        vm.startPrank(safeAddress);
+        vm.startStateDiffRecording();
+        new SphinxForkCheck{ salt: 0 }();
+        Vm.AccountAccess[] memory accountAccesses = vm.stopAndReturnStateDiff();
+
+        return checkAccesses(accountAccesses);
```

Comment:
We have to prank the safes address here or we get an error like the one in the screenshot below. I believe this is probably a bug in foundry triggered by the fact that I’m calling into `SphinxUtils` from `Sphinx` and then starting the state diff recording within `SphinxUtils`. I don’t think this will cause other problems for us, but just wanted to point it out in case you run into it.